tests/core_tests/chaingen.cpp

```diff
@@ -448,22 +447,40 @@ bool init_output_indices(map_output_idx_t& outs, std::map<uint64_t, std::vector<
         // TODO: add all other txes
         for (size_t i = 0; i < vtx.size(); i++) {
             const transaction &tx = *vtx[i];
+            const crypto::public_key& tx_pub_key = get_tx_pub_key_from_extra(tx);
+            const std::vector<crypto::public_key>& additional_tx_pub_keys = get_additional_tx_pub_keys_from_extra(tx);
+
+            bool is_key_derived = false;
+            bool is_key_derived_success = false;
+            crypto::key_derivation derivation;
 
             for (size_t j = 0; j < tx.vout.size(); ++j) {
                 const tx_out &out = tx.vout[j];
-
-                output_index oi(out.target, out.amount, boost::get<txin_gen>(*blk.miner_tx.vin.begin()).height, i, j, &blk, vtx[i]);
-                oi.set_rct(tx.version == 2);
-                oi.unlock_time = tx.unlock_time;
-                oi.is_coin_base = i == 0;
+                const txout_to_key & out_key = boost::get<txout_to_key>(out.target);
```

Comment:
This can now throw; line 461 was checking to make sure the variant had a `txout_to_key` object.

Reply:
Good catch. I'll move it below former 461, as there's not other reason for it to be up here other than some purism.